src/Output/Copy/Serializer/QuerySerializer.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types=1);
+
+namespace DoclerLabs\ApiClientGenerator\Output\Copy\Serializer;
+
+use DoclerLabs\ApiClientGenerator\Output\Copy\Request\RequestInterface;
+use DoclerLabs\ApiClientGenerator\Output\Copy\Schema\SerializableInterface;
+use JsonSerializable;
+
+class QuerySerializer
+{
+    public function serializeRequest(RequestInterface $request): string
+    {
+        $queryParameters = [];
+
+        foreach ($request->getRawQueryParameters() as $name => $value) {
+            if ($value === null) {
+                continue;
+            } elseif ($value instanceof SerializableInterface) {
+                $value = $value->toArray();
+            } elseif ($value instanceof JsonSerializable) {
+                $value = $value->jsonSerialize();
+            }
```

Comment:
we can later check instanceof DateTime or perhaps `format` from open api yaml definition, 'cause `date-time` means it's an ATOM string (https://swagger.io/docs/specification/data-models/data-types/). ATM generator/client doesn't "properly" support DateTime objects as query parameters.